### PR TITLE
SearchForm incl. API fetches

### PR DESCRIPTION
**Pull request by AnnekeSchuette**
Hi! This is my PR for the implementation of a search form and API connection services. It also contains some refactoring and outsourcing using custom hooks. Due to it's already large scope, the tests will come as separate PR. 😅 I look forward to your feedback and suggestions for improvement of all kinds! 🙏🏻 

## Value statement

As a **user** 
I **need** a **search form** to search **by dish, an ingredient, or a cuisine**
so that I can  **find** matching wines for my planned menu

### SearchForm
- the SearchForm contains an input field
- the submit button is disabled initially and form cannot be submitted by hitting enter
- a minimum of 3 characters typed is required until search button becomes active / form submit is possible
- enter or tap on search-button submits the form
### Results
- Results are rendered via WineListing component
- an empty result returns a message that nothing was found

### CODE REVIEW

File: client/src/components/WineCard/WineCard.jsx [hunk 1]
```diff
@@ -58,39 +60,41 @@ export default function WineCard({
...
+          </ListDescr>
         </DescrList>
       </CardInfo>
     </CardContent>
   )
 }
 
 const CardContent = styled.div`
-  background: #fff;
+  background: var(--color-ghost-white);
```
Comment (BerePendzialek) [inline client/src/components/WineCard/WineCard.jsx:91]: Really like the names you give to your colors ;)

File: client/src/services/useApi.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  const history = useHistory()
+
+  const { REACT_APP_API_KEY_SPOONACULAR } = process.env
+  const SPOONACULAR_API_BASEURL = 'https://api.spoonacular.com'
+
+  const getWinePairing = (food, opts) => {
+    const WINE_PAIRING_URL = `${SPOONACULAR_API_BASEURL}/food/wine/pairing?apiKey=${REACT_APP_API_KEY_SPOONACULAR}&food=${food}`
+
+    fetch(WINE_PAIRING_URL)
+      .then(res => res.json())
```
Comment (BerePendzialek) [inline client/src/services/useApi.js:16]: Happy to see you got the API. Maybe I can try it myself in a near future with the Spotify API.

File: client/src/components/WineCard/WineCard.jsx (hunk 1 above)
Comment (BerePendzialek) [inline client/src/components/WineCard/WineCard.jsx:91]: Really like the names you give to your colors ;)

File: client/src/components/Input/Input.stories.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+export default {
+  title: 'Capstone Project/Input',
+  component: Input,
+}
+
+const Template = args => <Input {...args} />
+
+export const Default = Template.bind({})
+Default.args = {
```
Comment (macfrei) [inline client/src/components/Input/Input.stories.js:11]: If you don't give an `id` here, then the input won't be connected to your label. (if you keep it like that)

File: client/src/components/SearchForm/SearchForm.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+export default function SearchForm({
+  onSubmit,
+  isDisabled,
+  search,
+  setSearch,
+}) {
+  return (
+    <SearchFormWrapper onSubmit={onSubmit}>
+      <Input
```
Comment (macfrei) [inline client/src/components/SearchForm/SearchForm.jsx:13]: If you don't give an `id` here, then the input won't be connected to your label. (if you keep it like that)

**Review by macfrei — COMMENTED** [review]
Good job! Added a few comments :)

File: client/src/components/App/App.jsx [hunk 5]
```diff
@@ -1,17 +1,24 @@
...
 import Header from 'components/Header/Header'
+import SearchForm from 'components/SearchForm/SearchForm'
 import WineListing from 'components/WineListing/WineListing'
 import WineStorage from 'components/WineStorage/WineStorage'
 import Navigation from 'components/Navigation/Navigation'
 import useToggleFavorite from 'hooks/useToggleFavorite'
 import usePageInfo from 'hooks/usePageInfo'
+import useApi from 'services/useApi'
+import useSearchForm from 'hooks/useSearchForm'
+import useLocalStorage from 'hooks/useLocalStorage'
```
Comment (marcushohlbein) [inline client/src/components/App/App.jsx:13]: nice with all your custom hooks

File: client/src/components/Button/Button.jsx [hunk 6]
```diff
@@ -34,23 +34,22 @@ export default function Button({
...
 
-  &.active {
-    background: rgb(215, 51, 163);
-    background: linear-gradient(
-      145deg,
-      rgba(215, 51, 163, 1) 18%,
-      rgba(53, 22, 178, 1) 100%
-    );
+  &.active,
+  :focus {
```
Comment (marcushohlbein) [inline client/src/components/Button/Button.jsx:47]: not quite sure, but is a focus on a mobile app needed?

File: client/src/components/SearchForm/SearchForm.spec.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+      render(<SearchForm />)
+    }
+  )
+  it('calls function onSubmit on form submit', () => {
+    const callback = jest.fn()
+    render(<SearchForm onSubmit={callback} />)
+    userEvent.click(screen.getByRole('button'))
+    expect(callback).toHaveBeenCalled()
+  })
+  it('updates on change', () => {
```
Comment (marcushohlbein) [inline client/src/components/SearchForm/SearchForm.spec.js:18]: very good test case

File: client/src/components/WineCard/WineCard.jsx [hunk 8]
```diff
@@ -58,39 +60,41 @@ export default function WineCard({
...
   box-shadow: 0px 1px 4px #00000030;
   color: var(--color-midnight);
-  border-radius: var(--space-small);
+  border-radius: var(--space-xxsmall);
```
Comment (marcushohlbein) [inline client/src/components/WineCard/WineCard.jsx:94]: xxsmall? 😅 do you mean xs?

File: client/src/components/WineCard/WineCard.jsx [hunk 9]
```diff
@@ -120,16 +124,15 @@ const BookmarkButton = styled.button`
 
   svg {
     fill: ${props =>
-      props.isActive ? 'var(--color-pink-pantone)' : 'inherit'};
+      props.isActive ? 'var(--color-candy-pink)' : 'var(--color-cadet-grey)'};
     stroke: 2px
       ${props =>
-        props.isActive ? 'var(--color-midnight)' : 'var(--color-pink-pantone)'};
+        props.isActive ? 'var(--color-cadet-grey)' : 'var(--color-candy-pink)'};
   }
```
Comment (marcushohlbein) [inline client/src/components/WineCard/WineCard.jsx:131]: awesome 💯

File: client/src/components/WineListing/WineListing.jsx [hunk 10]
```diff
@@ -1,48 +1,116 @@
...
+}) {
+  const { push } = useHistory()
+  const isValidResult =
+    !results.status && 'productMatches' in results && results.pairingText !== ''
+
+  const noResultsMessage =
+    !isValidResult &&
+    recentSearch !== '' &&
+    `Sorry, we couldn't find any matches for "${recentSearch}".😢
+    \n You could try another term or similar dish.`
```
Comment (marcushohlbein) [inline client/src/components/WineListing/WineListing.jsx:22]: well done frontend error handling 👍

**Review by marcushohlbein — COMMENTED** [review]
very good code. add a few comments in your code

File: client/src/components/Input/Input.jsx [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+  label,
+  name,
+  value,
+  className,
+  placeholder,
+  id,
+  onChange,
+}) {
+  return (
+    <InputWrapper className={'Input ' + className}>
```
[the remark this answers is not in the document]
Comment (AnnekeSchuette) [reply client/src/components/Input/Input.jsx:13]: Frankly for nothing 😅

File: client/src/components/Input/Input.stories.js (hunk 3 above)
Comment (AnnekeSchuette) [reply client/src/components/Input/Input.stories.js:11]: Thanks for the hint. It's easy to forget this property without looking into the Input component.

File: client/src/components/SearchForm/SearchForm.jsx (hunk 4 above)
Comment (AnnekeSchuette) [reply client/src/components/SearchForm/SearchForm.jsx:13]: Thanks for the hint. It's easy to forget this property without looking into the Input component.

File: client/src/components/Button/Button.jsx (hunk 6 above)
Comment (AnnekeSchuette) [reply client/src/components/Button/Button.jsx:47]: No, not really - I was just annoyed by how it was displayed in the browser. 😅

**Pull request merged**